north_tests/src/runtime.rs

```diff
@@ -14,98 +14,100 @@
 
 //! Controls North runtime instances
 
-use crate::process_assert::ProcessAssert;
-use color_eyre::eyre::{eyre, Error, Result, WrapErr};
+use color_eyre::eyre::{eyre, Result, WrapErr};
 use north::{api, runtime};
-use std::path::Path;
+use std::{future::Future, path::Path};
 use tokio::{time, time::timeout};
 
 const TIMEOUT: time::Duration = time::Duration::from_secs(3);
 
 /// A running instance of north.
 pub struct Runtime(runtime::Runtime);
 
+#[must_use = "Shoud be checked for expected response"]
+pub struct ApiResponse(Result<api::Response>);
+
+impl ApiResponse {
+    pub fn expect_ok(self) -> Result<()> {
+        match self.0 {
+            Ok(api::Response::Ok(())) => Ok(()),
+            _ => Err(eyre!("Response is not ok")),
+        }
+    }
+
+    pub fn expect_err(self, err: api::Error) -> Result<()> {
+        match self.0 {
+            Ok(api::Response::Err(e)) if err == e => Ok(()),
+            _ => Err(eyre!("Response is not an error")),
+        }
+    }
+
+    pub fn could_fail(self) {}
+}
+
+pub fn timeout_on<R>(f: impl Future<Output = R>) -> Result<R> {
+    futures::executor::block_on(timeout(TIMEOUT, f)).wrap_err("Future timed out")
```

Comment:
Why does this use the `futures` executor?

Reply:
I just wanted to get rid of all the  `await`s on the tests because I think there is no need for them to be asynchronous. Then I could not find a wait to do a `block_on` with `tokio`.